### DISCUSSION TITLE
Using and/or migrating to the new Akka.MultiNode.TestAdapter

### DISCUSSION THREAD

**Discussion by Arkatufus** [akkadotnet/akka.net#5482] [General]
`Akka.MultiNode.TestAdapter` is a new multi node spec/test runner that is replacing the old `Akka.MultiNodeTestRunner` package; it is based purely on the `Xunit` test framework and is compatible with `dotnet test`, Microsoft Visual Studio test runner, and JetBrains Rider test runner, enabling multi node tests to be run inside your IDE of choice.

To use the new runner, you will need to: 
1. Add two NuGet packages into your multi node test project:
  - [`Akka.MultiNode.TestAdapter`](https://www.nuget.org/packages/Akka.MultiNode.TestAdapter)
  - [`Microsoft.NET.Test.Sdk`](https://www.nuget.org/packages/Microsoft.NET.Test.Sdk)
2. Add an assembly level attribute to make sure that Xunit collection and parallelization are turned off:
```c#
[assembly: TestFramework("Akka.MultiNode.TestAdapter.MultiNodeTestFramework", "Akka.MultiNode.TestAdapter")]
``` 
3. Write your multi node tests. Please read [How to Write a Multi-Node Test](https://getakka.net/articles/testing/multi-node-testing.html#how-to-write-a-multi-node-test) in the documentation page.

To migrate to the new multi node test runner, you will need to: 
- Replace the old `Akka.MultiNodeTestRunner` package with the new `Akka.MultiNode.TestAdapter` package.
- Replace the old `MultiNodeFactAttribute` from the `Akka.Remote.TestKit` namespace to the new `MultiNodeFactAttribute` in the `Akka.MultiNode.TestAdapter` namespace. The easiest way to do this is to add a using statement in your multi node test file:
```C#
using MultiNodeFactAttribute = Akka.MultiNode.TestAdapter.MultiNodeFactAttribute;
```
- Add an assembly level attribute to make sure that Xunit collection and parallelization are turned off:
```c#
[assembly: TestFramework("Akka.MultiNode.TestAdapter.MultiNodeTestFramework", "Akka.MultiNode.TestAdapter")]
``` 

Why do we need the `TestFrameworkAttribute`?
`Akka.MultiNode.TestAdapter.MultiNodeTestFramework` is a custom `XunitTestFramework` implementation that ignores the Xunit Collection and Parallelization feature. This is done to make sure that each running tests are run in isolation from each other.

**Comment by Aaronontheweb** [comment]
Looks good to me @Arkatufus
